cache-noop/src/main/java/io/micronaut/cache/noop/NoOpCacheManager.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2017-2019 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.cache.noop;
+
+import io.micronaut.cache.DefaultCacheManager;
+import io.micronaut.cache.SyncCache;
+import io.micronaut.context.annotation.Primary;
+import io.micronaut.context.annotation.Replaces;
+
+import javax.annotation.Nonnull;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * A no operation {@link io.micronaut.cache.CacheManager} implementation suitable for disabling caching.
+ *
+ * @author Marcel Overdijk
+ * @since 1.0.0
+ */
+@Replaces(DefaultCacheManager.class)
+@Primary
+public class NoOpCacheManager implements io.micronaut.cache.CacheManager<Object> {
+
+    private Map<String, NoOpSyncCache> cacheMap;
+
+    /**
+     * Constructor.
+     */
+    public NoOpCacheManager() {
+        this.cacheMap = new HashMap<>();
```

Comment:
I think it would probably be better to use a `ConcurrentHashMap` here as concurrent access will results in exceptions which is not good even if for testing.

Reply:
Yes will do. Probably also good to do the same for the Ehcache and Hazelcast implementations as they use a `Map` as well.